File: longest-palindromic-substring.cpp
class Solution {
    
    public String longestPalindrome(String s) {
        //brute force
        if(s.length() < 2) return s;
        String res = "";
        for(int i=0; i<s.length();i++){
            for(int j=i; j<s.length(); j++){
                String current = s.substring(i, j+1);
                if(isPalindrome(current) && current.length()>res.length()) res = current;
            }
        }
        return res;
    }
    boolean isPalindrome(String s){
        int left = 0;
        int right = s.length()-1;
        while(left<=right){
            if(s.charAt(left)!=s.charAt(right)) return false;
            left++;
            right--;
        }
        return true;
    }
}

File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        int[] res = new int[] {0,0};
        for(int i = 0; i<s.length(); i++){
            int oddLength = expand(s, i, i);
            int evenLength = expand(s,i, i+1);
            if(oddLength > res[1] - res[0]+1){
                res[0] = i - oddLength/2;
                res [1] = i + oddLength/2;
            }
            if(evenLength > res[1] - res[0]+1){
                res[0] = i - evenLength/2 + 1;
                res[1] = i + (evenLength/2) ;

            }
        }
        return s.substring(res[0], res[1]+1);
    }
    public int expand(String s, int l, int r){
        while(l>=0 && r< s.length() && s.charAt(l)==s.charAt(r)){
            l--;
            r++;
        }
        return r-l-1;
    }
}